system/portenforce.py

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/python
+# encoding: utf-8
+
+# (c) 2016, Nathan Davison <ndavison85@gmail.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: portenforce
+author:
+    - '"Nathan Davison (@ndavison)" <ndavison85@gmail.com>'
+version_added: "2.1"
+description:
+    - Kill processes listening on TCP and UDP ports not defined in the whitelists.
+short_description: Kill processes listening on TCP and UDP ports not defined in the whitelists.
+options:
+  whitelist_tcp:
+    description:
+      - A list of TCP ports to allow.
+    required: false
+    type: "list"
+  whitelist_udp:
+    description:
+      - A list of UDP ports to allow.
+    required: false
+    type: "list"
+'''
+
+EXAMPLES = '''
+- name: only allow TCP 80, 443, and 22
+  portenforce:
+    whitelist_tcp:
+      - 22
+      - 80
+      - 443
+'''
+
+RETURN = '''
+killed:
+  description: A list of pids that were killed.
+  type: list
+  contains:
+    pid:
+      description: The pid of a killed process.
+      type: int
+      sample: 1223
+    port:
+      description: The port the killed pid was listening on.
+      type: int
+      sample: 443
+    proto:
+      description: The protocol, TCP or UDP, of the listening connection.
+      type: string
+      sample: 'tcp'
+'''
+
+import re
+from subprocess import Popen, PIPE
+
+def netStatParse(raw):
+    results = list()
+    for line in iter(raw.splitlines()):
+        listening_search = re.search('[^ ]+:[^ ]+', line)
+        if listening_search:
+            splitted = line.split()
+            conns = re.search('[^ ]+:([^ ]+)', splitted[3])
+            if 'tcp' in splitted[0]:
+                proto = 'tcp'
+                pidstr = splitted[6]
+            else:

```

Comment:
If something is neither tcp or udp, this would be classified as udp, that's likely not what you want. It would be ok because there is previous filtering since we only look on a specific option of netstat, but I would suggest to be more defensive in this part of the code.